homeassistant/components/rtorrent/__init__.py

```diff
@@ -1 +1,182 @@
 """The rtorrent component."""
+from datetime import timedelta
+import logging
+import xmlrpc.client
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_MONITORED_VARIABLES,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+    CONF_URL,
+    DATA_RATE_KILOBYTES_PER_SECOND,
+    DATA_RATE_MEGABYTES_PER_SECOND,
+    STATE_IDLE,
+)
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.event import track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_SPEED = "speed"
+
+DOMAIN = "rtorrent"
+DATA_RTORRENT = "data_rtorrent"
+DATA_UPDATED = "rtorrent_data_updated"
+
+DEFAULT_NAME = "rTorrent"
+DEFAULT_SPEED_LIMIT = 1000  # 1 MB/s
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=5)
+
+SENSOR_TYPES = {
+    "current_status": ["Status", None, "mdi:server-network"],
+    "upload_speed": ["Up Speed", DATA_RATE_MEGABYTES_PER_SECOND, "mdi:upload"],
+    "download_speed": ["Down Speed", DATA_RATE_MEGABYTES_PER_SECOND, "mdi:download"],
+    "upload_limit": ["Up Limit", DATA_RATE_KILOBYTES_PER_SECOND, "mdi:upload"],
+    "download_limit": ["Down Limit", DATA_RATE_KILOBYTES_PER_SECOND, "mdi:download"],
+    "all_torrents": ["All Torrents", None, "mdi:view-list-outline"],
+    "stopped_torrents": ["Stopped Torrents", None, "mdi:stop"],
+    "complete_torrents": ["Complete Torrents", None, "mdi:playlist_check"],
+    "uploading_torrents": ["Uploading Torrents", None, "mdi:upload-multiple"],
+    "downloading_torrents": ["Downloading Torrents", None, "mdi:download-multiple"],
+    "active_torrents": ["Active Torrents", None, "mdi:playlist-play"],
+}
+
+CONFIG_SCHEMA = vol.Schema(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler